### PR TITLE
HEEDLS-440 - Implemented course content page

### PR DESCRIPTION
**Pull request by DanBloxham-sw**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-440

### Description
Implemented new Course Content page. I believe everything in the ticket has been completed.

### Screenshots
Desktop:
Full desktop

Mobile:

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [x] Updated/added documentation in Swiki and/or Readme.
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/TutorialContentDataService.cs [hunk 1]
```diff
@@ -275,5 +278,22 @@ FROM CustomisationTutorials
...
+        {
+            return connection.Query<Tutorial>(
+                @"SELECT 
+                        tu.TutorialID,
+                        tu.TutorialName,
+                        ct.[Status],
+                        ct.DiagStatus
+                    FROM dbo.Tutorials AS tu
+                    LEFT JOIN dbo.CustomisationTutorials AS ct ON ct.TutorialID = tu.TutorialID
+                        AND ct.CustomisationID = @customisationId
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Data/DataServices/TutorialContentDataService.cs:292]: Formatting

File: DigitalLearningSolutions.Data/Services/SectionService.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    using System.Linq;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models;
+
+    public interface ISectionService
+    {
+        IEnumerable<Section> GetSectionsAndTutorialsForCustomisation(int customisationId, int applicationId);
+    }
+
+
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Data/Services/SectionService.cs:14]: Extra line

File: DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseContentControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using DigitalLearningSolutions.Web.Controllers.TrackingSystem.CourseSetup;
+    using DigitalLearningSolutions.Web.Tests.ControllerHelpers;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup.CourseContent;
+    using FakeItEasy;
+    using NUnit.Framework;
+    using FluentAssertions.AspNetCore.Mvc;
+
+
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseContentControllerTests.cs:15]: Extra line

File: DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseContentControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            // Then
+            result.Should().BeNotFoundResult();
+        }
+
+        [Test]
+        public void Index_returns_Index_page_when_appropriate_course_found()
+        { 
+            // Given
+            A.CallTo(() => courseDataService.GetCourseDetails(A<int>._, A<int>._, A<int>._)).Returns(CourseDetailsTestHelper.GetDefaultCourseDetails());
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/CourseContentControllerTests.cs:47]: Long line. Run fomatter?

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/CourseSetup/CourseContent/CourseContentViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup.CourseContent
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Models;
+
+    public class CourseContentViewModel
+    {
+        public CourseContentViewModel(int customisationId, string courseName, bool postLearningAssessment, IEnumerable<Section> sections)
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Web/ViewModels/TrackingSystem/CourseSetup/CourseContent/CourseContentViewModel.cs:9]: Long line? formatter

**Review by AlexJacksonDS — COMMENTED** [review]
Overall looks fine to me, a few formatting issues.

Note that this ticket has been handed over to me to do any markups on while Dan is away.

**Review by JBsoftwire — APPROVED** [review]
This looks fine overall to me. I made a couple of non-essential comments you can decide if you want to deal with when merging, they aren't worth a re-review.

**Pull request merged**